### PR TITLE
feat: Add SNS topic delivery status IAM role

### PR DESCRIPTION
**Pull request by chris3ware**
## Description

The purpose of this pull request is to add delivery status logging for the SNS topic. This is achieved by specifying an IAM role with permission to write to CloudWatch logs. If a role is not specified, then a role will be created - if delivery logging is required.

## Motivation and Context

I recently worked on a project which required compliance with [NCSC Operational Best Practices](https://docs.aws.amazon.com/config/latest/developerguide/operational-best-practices-for-ncsc.html). SNS delivery status logging must be enabled to comply with this framework. I created the topic outside of the module initially, but thought it would be a useful addition to the module.

## How Has This Been Tested?
- :white_check_mark: I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- :white_check_mark: I have tested and validated these changes using one or more of the provided `examples/*` projects

- :white_check_mark: I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

File: outputs.tf [hunk 1]
```diff
@@ -48,3 +48,8 @@ output "lambda_cloudwatch_log_group_arn" {
   description = "The Amazon Resource Name (ARN) specifying the log group"
   value       = try(aws_cloudwatch_log_group.lambda[0].arn, "")
 }
+
+output "sns_topic_feedback_role_arn" {
+  description = "The Amazon Resource Name (ARN) of the IAM role used for SNS delivery status logging"
+  value       = try(local.sns_feedback_role, "")
```
Comment (antonbabenko) [inline outputs.tf:54]:
```suggestion
  value       = local.sns_feedback_role
```

**Review by antonbabenko — APPROVED** [review]
LGTM (minor comment), please rebase, and let's merge it :) Thank you!

**Comment by chris3ware** [comment]
Hi, is anything else required from me to merge this?

**Comment by antonbabenko** [comment]
This PR is included in version 5.5.0 :tada:

**Pull request merged**